### ISSUE TITLE
Additional control over metrics logging

### ISSUE THREAD

**Issue by pietrolesci** [PyTorchLightning/lightning-flash#452]
## 🚀 Feature
When defining a `Task`, like `TextClassifier`, allow the user to either:
- decide how to log the metrics (e.g., avoid logging to progress bar) _globally_, or
- decide how to log each individual metric (e.g., log accuracy to progress bar but avoid logging any non-scalar one)

### Motivation
When logging metrics from `torchmetrics` like `ConfusionMatrix`, logging to progress bar creates errors. Flash automatically logs to progress bar, for example [here](https://github.com/PyTorchLightning/lightning-flash/blob/2ec52e633bb3679f50dd7e30526885a4547e1851/flash/core/model.py#L197), and this forces the user to subclass and redefine the `{train, val, test}_steps` only to modify how logging works.
